### ISSUE TITLE
⚠️ `fr-staging` is down. Could not start postgresql // `could not load server certificate file "/etc/postgresql/fullchain.pem": No such file or directory`

### ISSUE THREAD

**Issue by jibees** [openfoodfoundation/ofn-install#854]
When provisioning `fr-staging` everything seems fine:
```
TASK [db_integrations : Install script to copy valid TLS certificates for postgresql] ************************************************************************************************
Vendredi 30 décembre 2022  15:54:15 +0100 (0:00:00.052)       0:01:10.804 *****
changed: [staging.coopcircuits.fr]

TASK [db_integrations : Call copy script on certificate renewal] *********************************************************************************************************************
Vendredi 30 décembre 2022  15:54:17 +0100 (0:00:02.587)       0:01:13.391 *****
changed: [staging.coopcircuits.fr]

TASK [db_integrations : Use valid TLS certificate] ***********************************************************************************************************************************
Vendredi 30 décembre 2022  15:54:18 +0100 (0:00:01.233)       0:01:14.625 *****
changed: [staging.coopcircuits.fr]

TASK [db_integrations : Use valid TLS key] *******************************************************************************************************************************************
Vendredi 30 décembre 2022  15:54:20 +0100 (0:00:01.243)       0:01:15.869 *****
changed: [staging.coopcircuits.fr]

RUNNING HANDLER [restart postgres] ***************************************************************************************************************************************************
Vendredi 30 décembre 2022  15:54:21 +0100 (0:00:01.255)       0:01:17.124 *****
changed: [staging.coopcircuits.fr]

RUNNING HANDLER [geerlingguy.postgresql : restart postgresql] ************************************************************************************************************************
Vendredi 30 décembre 2022  15:54:25 +0100 (0:00:04.438)       0:01:21.563 *****
[WARNING]: Ignoring "sleep" as it is not used in "systemd"
changed: [staging.coopcircuits.fr]

PLAY RECAP ***************************************************************************************************************************************************************************
staging.coopcircuits.fr    : ok=36   changed=10   unreachable=0    failed=0    skipped=21   rescued=0    ignored=0
```

But now postgresql server is down:

```
ofn-admin@vps-2a25f087:~$ pg_lsclusters
Ver Cluster Port Status Owner    Data directory              Log file
10  main    5432 down   postgres /var/lib/postgresql/10/main /var/log/postgresql/postgresql-10-main.log
13  main    5433 online postgres /var/lib/postgresql/13/main /var/log/postgresql/postgresql-13-main.log
```
```
Dec 30 15:05:24 vps-2a25f087 postgresql@10-main[15015]: Error: /usr/lib/postgresql/10/bin/pg_ctl /usr/lib/postgresql/10/bin/pg_ctl start -D /var/lib/postgresql/10/main -l /var/log/postgresql/postgresql-10-main.log -s -o  -c config_file="/et
Dec 30 15:05:24 vps-2a25f087 postgresql@10-main[15015]: 2022-12-30 15:05:24.074 UTC [15020] FATAL:  could not load server certificate file "/etc/postgresql/fullchain.pem": No such file or directory
```

_Originally posted by @jibees in https://github.com/openfoodfoundation/ofn-install/issues/853#issuecomment-1367957737_

**Comment by filipefurtad0** [comment]
Just noting that staging-UK is down too, due to my staging attempt:
https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/uk-staging/deploys/1496

**Comment by jibees** [comment]
for the same reason I guess...

```
fatal: [staging.openfoodnetwork.org.uk]: FAILED! => {"changed": false, "cmd": "/bin/gzip </var/lib/postgresql/.ansible/tmp/ansible-moduletmp-1672406963.05-SI6YEZ/pg_fifo > /var/lib/postgresql/rollback.sql.gz & /usr/bin/pg_dump openfoodnetwork --port=5432 --username=postgres >/var/lib/postgresql/.ansible/tmp/ansible-moduletmp-1672406963.05-SI6YEZ/pg_fifo", "msg": "pg_dump: [archiver (db)] connection to database \"openfoodnetwork\" failed: could not connect to server: No such file or directory\n\tIs the server running locally and accepting\n\tconnections on Unix domain socket \"/var/run/postgresql/.s.PGSQL.5432\"?\n", "rc": 1, "stdout": "", "stdout_lines": []}
```

**Comment by jibees** [comment]
I've executed the following commands on `fr-staging`:

```
root@vps-2a25f087:~# cp /etc/letsencrypt/live/staging.coopcircuits.fr/fullchain.pem /etc/postgresql/
root@vps-2a25f087:~# cp /etc/letsencrypt/live/staging.coopcircuits.fr/privkey.pem /etc/postgresql/
root@vps-2a25f087:~# chown postgres:postgres /etc/postgresql/*.pem
```
Which is basically what `Install script to copy valid TLS certificates for postgresql` step should have done via `roles/db_integrations/templates/cert-renewal-hook.sh.j2` script. 

After restarting, this seems way much better:
```
ofn-admin@vps-2a25f087:~$ pg_lsclusters
Ver Cluster Port Status Owner    Data directory              Log file
10  main    5432 online postgres /var/lib/postgresql/10/main /var/log/postgresql/postgresql-10-main.log
13  main    5433 online postgres /var/lib/postgresql/13/main /var/log/postgresql/postgresql-13-main.log
```

**Comment by jibees** [comment]
Unfortunately, even if postgresql is up and running, can't deploy on staging. This time, seems to be due to password miss-configuration for `ofn_user`:

```
 FATAL:  password authentication failed for user \"ofn_user\"\nFATAL:  password authentication failed for user \"ofn_user\"
```

**Comment by jibees** [comment]
@mkllnk If you could have a look, I will be grateful to you ;) 🙏

**Comment by mkllnk** [comment]
I found a mistake in https://github.com/openfoodfoundation/ofn-install/pull/853 which caused the missing certificate error. You solved that manually but the app couldn't connect to the database. It had the wrong password. Maybe someone provisioned with the wrong secrets file? Or pasted the wrong password manually?

I provisioned the app (`-t app`) again and the database config has the right password now. I can start the Rails console and the app is working again. :heavy_check_mark:
